### PR TITLE
Initialize Jenkins client which comes from jenkins-zh/jenkins-cli

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR dose

- [x] Initialize Jenkins client which comes from [jenkins-zh/jenkins-cli](https://github.com/jenkins-zh/jenkins-cli/)
- [x] Add support for GitHub action
- [x]  Provide Makefile

### Why we need it

If we separated Jenkins client from [Jenkins cli](https://github.com/jenkins-zh/jenkins-cli/), other projects outside would **lightly** introduce Jenkins client and interact with Jenkins easily.

### CODE REVIEW

File: artifacts.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package jenkins_client
```
Comment (LinuxSuRen) [inline artifacts.go:1]: I prefer to use `pkg` as the name of the package. `jenkins_client` is too long.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: artifacts.go (hunk 1 above)
Comment (JohnNiang) [reply artifacts.go:1]: Considering Jenkins client will be used by others, I moved `client/*` to /*

Comment (LinuxSuRen) [reply artifacts.go:1]: How about move those files into the directory `pkg`?

Comment (JohnNiang) [reply artifacts.go:1]: That's great.

File: pkg/artifacts.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package client
```
Comment (LinuxSuRen) [inline pkg/artifacts.go:1]: ? Why client  instead of pkg?

Comment (JohnNiang) [reply pkg/artifacts.go:1]: I think `pkg` is not a good name for package and has not relationship with Jenkins client.

Comment (JohnNiang) [reply pkg/artifacts.go:1]: Please see https://github.com/minio/minio-go

Comment (LinuxSuRen) [reply pkg/artifacts.go:1]: IMO, `client` is not a good name. All source files of this project belong to a client. How about creating some sub-directory. Such as:

```
pkg/job
...
pkg/artifact
...
pkg/credential
```

See also the [Golang package best practices](https://rakyll.org/style-packages/).

**Comment by JohnNiang** [comment]
hi @LinuxSuRen , the GitHub Action dose not work for this PR. Please have a check for action settings of `jenkins-client` repository.

**Comment by JohnNiang** [comment]
For GitHub checks result, please check [here](https://github.com/JohnNiang/jenkins-client/runs/3397479147?check_suite_focus=true).

**Review by LinuxSuRen — APPROVED** [review]

**Pull request merged**